homework01/test/examples/Main.hs

```diff
@@ -1,15 +1,44 @@
-module Main
-  ( main
-  )
-  where
-
--- doctest
-import qualified Test.DocTest as DocTest
+module Main where
 
+import Basis
+import Test.Hspec
 
 main :: IO ()
-main =
-  DocTest.doctest
-    [ "-isrc"
-    , "src/Basis.hs"
-    ]
+main = hspec $ do
+
+  describe "toDigits" $ do
+    it "converts positive integers to a list of digits" $
+      toDigits 1234 `shouldBe` [1, 2, 3, 4]
+
+    it "returns an empty list for 0" $
+      toDigits 0 `shouldBe` []
+
+    it "returns an empty list for negative inputs" $
+      toDigits (-17) `shouldBe` []
+
+  describe "toDigitsRev" $
+    it "converts positive integers to a list of digits reversed" $
+      toDigitsRev 1234 `shouldBe` [4, 3, 2, 1]
+
+  describe "doubleEveryOther" $ do
```

Comment:
I'm not sure if it's necessary to add a test with an empty lists as input, because the exercise doesn't indicate this kind of input. I know that is necessary to test all cases (valid and invalid inputs) but this can produce confusion in a Haskell beginner when he test `doubleEveryOther` function and don't pass a test with an empty list.